lib/azure/armrest/storage_account_service.rb

```diff
@@ -28,6 +28,33 @@ def list_all(filter = {})
         super(filter).each { |model| model.configuration = configuration }
       end
 
+      # Retrieve information about the storage account using a URI. You may optionally
+      # specify a resource group name, which will slightly faster. By default it will
+      # use the resource group specified in the service configuration, if any.
+      #
+      # Example:
+      #
+      #   sas = Azure::Armrest::StorageAccountService.new(<config>)
+      #   url = https://foo.blob.core.windows.net/vhds/dberger1201691213010.vhd"
+      #
+      #   sas.get_from_url(url)
+      #   sas.get_from_url(url, some_resource_group)
+      #
+      def get_from_url(url, resource_group = configuration.resource_group)
+        uri  = Addressable::URI.parse(url)
+        name = uri.host.split('.').first
+
+        unless resource_group
+          rservice = Azure::Armrest::ResourceService.new(configuration)
+          filter   = "resourceType eq 'Microsoft.Storage/storageAccounts' and name eq '#{name}'"
+          resource = rservice.list_all(:filter => filter, :all => true).first
+          raise ArgumentError, "unable to find resource group for #{url}" unless resource
+          resource_group = resource.id[/resourceGroups\/(.*?)\//i, 1]
+        end
+
+        get(name, resource_group)
```

Comment:
@djberg96 - Do you need to set the configuration like you did before in the `get_from_vm` method? Like this:
`get(name, resource_group).configuration = configuration` ?

Reply:
@bronaghs No, it will already have that information.